service/handlers.go

```diff
@@ -1002,14 +1100,17 @@ func (h handler) listTokens(w http.ResponseWriter, r *http.Request) {
 
 	ctx := r.Context()
 
-	_, err = h.dbClient.ReadProjectEntry(ctx, projectName)
+	level.Debug(l).Log("message", "creating credential provider")
+	cp, err := h.newCredentialsProvider(*a, h.env, r.Header, credentials.NewVaultConfig, credentials.NewVaultSvc)
 	if err != nil {
-		level.Error(l).Log("message", "error retrieving project", "error", err)
-		if errors.Is(err, upper.ErrNoMoreRows) {
-			h.errorResponse(w, "project does not exist", http.StatusNotFound)
-		} else {
-			h.errorResponse(w, "error retrieving project", http.StatusInternalServerError)
-		}
+		level.Error(l).Log("message", "error creating credentials provider", "error", err)
+		h.errorResponse(w, "error creating credentials provider", http.StatusInternalServerError)
+		return
+	}
+
+	projectExists, err := h.projectExists(ctx, l, cp, w, projectName)
+
+	if err != nil || !projectExists {
```

Comment:
Should we at least be logging error here instead of swallowing it entirely?

Reply:
the logging is taking place in the projectExists method